a-short-introduction-to-kubernetes/index.md

```diff
@@ -0,0 +1,372 @@
+
+I will do a walk through of some of the basic pieces of the Kubernetes puzzle, and what they mean in a Directus context.
+
+:::info Please note
+
+The example files in this post, should not be used to create a Directus setup, they are just there to describe the different parts, please see the GitHub gist in the end of the post if you are interested in a full working example.
+
+:::
+
+## Yaml and API:s
+
+All objects can be described and created with yaml, with a minimum requirement telling which api you are going to use, and what kind, like:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: directus
+  labels:
+    app: directus
+spec:
+  containers:
+  - name: directus
+    image: directus/directus:10.8.3
+    ports:
+      - name: http
+        containerPort: 8055
+        protocol: TCP
+```
+
+## Metadata and labels
+
+Metadata is used to identify your Kubernetes objects, without them you soon will be lost, and even different parts of the Kubernetes puzzle requires them. Some metadata is a must, like the name - otherwise there is no identification. Labels and annotations adds metadata to your objects so you could easily find them or target them.
+
+One of the reasons is then you create an object, like a Deployment, a random string is used for the pod it creates. So you to communicate with it, you need the metadata to identify it.
+
+Another, extremely useful metadata is annotations, but that is out of the scope for this post.
+
+## Containers
+
+Containers are the Docker images you deploy to your K8s cluster. There are mainly two kind of containers, initContainers and containers. initContainers runs before your containers start, like if you need to set permissions, or do some task like updating your Directus schema etc.
+
+## Environment variables
+
+Environment variables could be added as part of your object, like the public url for Directus.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: directus-app
+  labels:
+    app: directus
+spec:
+  containers:
+  - name: directus
+    image: directus/directus:10.8.3
+    ports:
+    - containerPort: 80
+    env:
+      - name: PUBLIC_URL
+        value: https://directus.app
+```
+
+You could also use Secrets and use with for setting environment variables, that I will not cover here.
+
+## Pods
+
+A pod is the smallest deployable compute object in K8s, like the example above. A pod is what is running one or several docker containers, a pod could be Directus, MySQL or Redis. If your pod crashes, it's dead and doesn't restart - it's like when you start a docker container with `docker run directus` and suddenly something gets weird, or your resources run out. So you need something to handle pods with, and the logic they are started, with config and volumes, there is where Deployments and StatefulSets come in.
```

Comment:
This reads great. Great examples, caveats, and lead-in to the next bit of knowledge.